src/lib/yaraparse.py

```diff
@@ -186,44 +235,63 @@ def flatten_regex_or_tree(unit: Any) -> Optional[List[bytes]]:
         return None
 
 
+def urisfy_regex_or_tree(
+    unit: Any, is_ascii: bool, is_wide: bool, is_nocase: bool,
+) -> Optional[UrsaExpression]:
+    or_strings = flatten_regex_or_tree(unit)
+    if or_strings and all(s is not None for s in or_strings):
+        or_ursa_strings: List[UrsaExpression] = []
+        for s in or_strings:
+            ursified_s = ursify_regex_text(
+                s, is_ascii=is_ascii, is_wide=is_wide, is_nocase=is_nocase
+            )
+            if ursified_s:
```

Comment:
I'm wondering if the `else` statement here shouldn't lead to early return as well.
My logic would be that:
* for `and` expressions, we can drop non-parsable strings as this will only lead to possible false positives
* for `or` expressions, we either have to correctly parse all strings or none at all, as dropping even a single one can lead to a false negative

Reply:
Great point. Fixed and I've also added a test for this case. I hope you don't mind me just adding more rules to existing test files. Let me know if you want it separated.